### PR TITLE
DTSPO-9662: migrate repo to cnp template

### PR DESCRIPTION
**Pull request by narinder17**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-9662

### Change description ###
Updated the stages for plan & apply to run new pipeline 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ variables:
   - name: timeoutInMinutes
     value: "60"
   - name: agentPool
-    value: "ubuntu-latest"
+    value: "ubuntu-18.04"
```
Comment (madjava) [inline azure-pipelines.yml:22]: kindly revert back to latest 👌🏾

**Comment by madjava** [comment]
@narinder17 verify that the state file in new location, if one was created, has the contents of the statefile in old location. There should be nothing to add of create once done.

**Comment by madjava** [comment]
closed my mistake, sorry

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,9 +1,11 @@
+ # trigger master branch
 trigger:
   batch: true
   branches:
     include:
       - master
 
+# repo to use - cnp
```
Comment (thomast1906) [inline azure-pipelines.yml:8]: remove unnecessary comments

File: azure-pipelines.yml [hunk 3]
```diff
@@ -12,213 +14,77 @@ resources:
     name: hmcts/cnp-azuredevops-libraries
     endpoint: 'hmcts'
 
+# cnp-azuredevops-libraries
 variables:
-  agentPool: 'ubuntu-18.04'
-  product: 'cft-platform'
-
+  - name: timeoutInMinutes
+    value: "60"
```
Comment (thomast1906) [inline azure-pipelines.yml:20]: is this needed? default is 60

File: environments/ithc/ithc-platform-hmcts-net.yml [hunk 4]
```diff
@@ -62,6 +62,9 @@ A:
     - 10.143.33.133
     ttl: 300
 cname:
+  - name: c100-application
```
Comment (thomast1906) [inline environments/ithc/ithc-platform-hmcts-net.yml:65]: curious on the additions of multiple dns records?

**Comment by timja** [comment]
conflicts with master

**Review by timja — CHANGES_REQUESTED** [review]
why are only 3 environments included?

Please improve the stage names:

image

File: azure-pipelines.yml (hunk 1 above)
Comment (vijay-tenoy) [reply azure-pipelines.yml]: Done

File: azure-pipelines.yml (hunk 3 above)
Comment (vijay-tenoy) [reply azure-pipelines.yml]: Removed timeout

**Comment by vijay-tenoy** [comment]
> why are only 3 environments included?
> 
> Please improve the stage names:
> 
> image

All environments re-introduced and [build passing](https://dev.azure.com/hmcts/CNP/_build/results?buildId=266158&view=results) with terraform plan having no changes in any environment

File: azure-pipelines.yml [hunk 5]
```diff
@@ -13,212 +13,113 @@ resources:
...
+  - template: vars/input-variables.yaml@cnp-azuredevops-libraries
+  
+parameters:
+  - name: overrideAction
+    type: string
+    default: plan
+    values:
+      - plan
+      - apply
+      - destroy
```
Comment (timja) [inline azure-pipelines.yml:33]: we don't want destroy here
```suggestion
```

File: environments/prod/mailrelay-platform-hmcts-net.yml [hunk 6]
```diff
@@ -34,7 +34,4 @@ A:
 cname:
   - name: "dev-in"
     ttl: 60
-    record: "ss-dev-mailrelay.trafficmanager.net"
-  - name: "prod-in"
-    ttl: 60
-    record: "ss-prod-mailrelay.trafficmanager.net"
```
Comment (timja) [inline environments/prod/mailrelay-platform-hmcts-net.yml:40]: why is this getting removed here?

File: environments/staging/staging.tfvars [hunk 7]
```diff
@@ -0,0 +1 @@
+env = "aat"
```
Comment (timja) [inline environments/staging/staging.tfvars:1]: why does this need setting now and not before? (as far as I can tell)

File: environments/prod/mailrelay-platform-hmcts-net.yml (hunk 6 above)
Comment (madjava) [reply environments/prod/mailrelay-platform-hmcts-net.yml:40]: I asked him to remove it, clashing with the A record, i'll circle back to mailrelay next spring. The branch fails on merge if this is in

Comment (timja) [reply environments/prod/mailrelay-platform-hmcts-net.yml:40]: should be removed in a separate PR though and not this one

File: environments/staging/staging.tfvars (hunk 7 above)
Comment (vijay-tenoy) [reply environments/staging/staging.tfvars:1]: This is the way it is in [master](https://github.com/hmcts/azure-private-dns/blob/1e7b4119cbea3206ddd327a723850f6a4fa7160c/components/staging/variables.tf#L7). Just re-adding it back.  Another alternative will be to add `staging` in common tag module [on this line](https://github.com/hmcts/terraform-module-common-tags/blob/093dcb24e2d1973945b093dc31506222fda0c9c0/main.tf#L45)

File: environments/prod/mailrelay-platform-hmcts-net.yml (hunk 6 above)
Comment (madjava) [reply environments/prod/mailrelay-platform-hmcts-net.yml:40]: fair point, i am just looking for a clean build on merge this time for this repo. 👌🏾

**Comment by madjava** [comment]
@vijay-tenoy noticed some empty tfvars files in the environment folder, are they needed if not please delete them 💪🏾

File: environments/staging/staging.tfvars (hunk 7 above)
Comment (timja) [reply environments/staging/staging.tfvars:1]: or rename the folder to be stg

**Review by timja — APPROVED** [review]

**Pull request merged**